From: Jens Axboe <axboe@fb.com>
Date: Wed, 7 Dec 2016 15:53:18 -0700
Subject: blk-flush: run the queue when inserting blk-mq flush
References: bsc#1020989,FATE#322379
Git-commit: c8e52ba5e2d6df5acaaeedda20d74f4ec3adcc82
Patch-Mainline: v4.10-rc1

Currently we pass in to run the queue async, but don't flag the
queue to be run. We don't need to run it async here, but we should
run it. So fixup the parameters.

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Hannes Reinecke <hare@suse.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-flush.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-flush.c b/block/blk-flush.c
index 0a3cf9a..3444d99 100644
--- a/block/blk-flush.c
+++ b/block/blk-flush.c
@@ -454,7 +454,7 @@ void blk_insert_flush(struct request *rq)
 	if ((policy & REQ_FSEQ_DATA) &&
 	    !(policy & (REQ_FSEQ_PREFLUSH | REQ_FSEQ_POSTFLUSH))) {
 		if (q->mq_ops) {
-			blk_mq_insert_request(rq, false, false, true);
+			blk_mq_insert_request(rq, false, true, false);
 		} else
 			list_add_tail(&rq->queuelist, &q->queue_head);
 		return;
-- 
1.8.5.6

